csc.dll exited with code 1

_From @ryanbrandenburg on February 9, 2018 19:58_

## Steps to reproduce
1. dotnet new mvc
2. Add `.AddFormatterMappings(m => {);` between `.AddMvc()` and `.SetCompatibilityVersion` in 'Startup.cs'.
3. dotnet run

## Expected  behavior
I would expect to get an informative error message about the mistakes I've just made (leaving off a trailing "}" and adding a ";", then trying to call `.SetCompatibilityVersion`.).

## Actual behavior
It returns:
```
/home/rybrande/dotnet/sdk/2.1.300-preview1-008030/Roslyn/Microsoft.CSharp.Core.targets(84,5): error MSB6006: "csc.dll" exited with code 1. [/home/rybrande/functional/src/Fail/Fail.csproj]

The build failed. Please fix the build errors and run again.
```
with no other identifying or helpful information.

## Environment data
`dotnet --info` output:
```
.NET Command Line Tools (2.1.300-preview1-008030)

Product Information:
 Version:            2.1.300-preview1-008030
 Commit SHA-1 hash:  be228a492f

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  16.04
 OS Platform: Linux
 RID:         ubuntu.16.04-x64
 Base Path:   /home/rybrande/dotnet/sdk/2.1.300-preview1-008030/

Microsoft .NET Core Shared Framework Host

  Version  : 2.1.0-preview1-26207-04
  Build    : f2c3216183d20416568a4bbf5bb7d153e826f153
```

Ubuntu 16.04

Found while doing aspnet/Release#215.

_Copied from original issue: dotnet/cli#8581_

I agree that the compiler is providing the error message, but it doesn't look like it's getting show to the user. Here's what I see when running the repro using the latest SDK and recording the build output to a file log:

        Using shared compilation with compiler from directory: /Users/angocke/tmp/dotnet-sdk-latest/sdk/2.1.300-preview2-008171/Roslyn/bincore
         Microsoft (R) Visual C# Compiler version 2.8.0.62608 (b2a417c1)
         Copyright (C) Microsoft Corporation. All rights reserved.
         
         Startup.cs(32,58): error CS1513: } expected
         Startup.cs(32,60): error CS1513: } expected
   1:7>/Users/angocke/tmp/dotnet-sdk-latest/sdk/2.1.300-preview2-008171/Roslyn/Microsoft.CSharp.Core.targets(84,5): error MSB6006: "csc.dll" exited with code 1. [/Users/angocke/tmp/test24736/test24736.csproj]
       Done executing task "Csc" -- FAILED.

--- a/src/Compilers/Core/MSBuildTask/Csc.cs
+++ b/src/Compilers/Core/MSBuildTask/Csc.cs
@@ -147,7 +147,7 @@ namespace Microsoft.CodeAnalysis.BuildTasks
 
         #region Tool Members
 
-        private static readonly string[] s_separators = { "\r\n" };
+        private static readonly string[] s_separators = { Environment.NewLine };
 
         internal override void LogMessages(string output, MessageImportance messageImportance)
         {

--- a/src/Compilers/Core/MSBuildTask/ManagedCompiler.cs
+++ b/src/Compilers/Core/MSBuildTask/ManagedCompiler.cs
@@ -625,7 +625,7 @@ namespace Microsoft.CodeAnalysis.BuildTasks
 
         internal static void LogErrorOutput(string output, TaskLoggingHelper log)
         {
-            string[] lines = output.Split(new string[] { "\r\n" }, StringSplitOptions.RemoveEmptyEntries);
+            string[] lines = output.Split(new string[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
             foreach (string line in lines)
             {
                 string trimmedMessage = line.Trim();

--- a/src/Compilers/Core/MSBuildTask/Vbc.cs
+++ b/src/Compilers/Core/MSBuildTask/Vbc.cs
@@ -227,7 +227,7 @@ namespace Microsoft.CodeAnalysis.BuildTasks
 
         #region Tool Members
 
-        private static readonly string[] s_separator = { "\r\n" };
+        private static readonly string[] s_separator = { Environment.NewLine };
 
         internal override void LogMessages(string output, MessageImportance messageImportance)
         {

